### PR TITLE
Update Compose reference API to v2.16.0

### PR DESCRIPTION
**Pull request by glours**
Signed-off-by: GitHub <noreply@github.com>

### Proposed changes

Replace auto-generated PR #16692

### CODE REVIEW

**Comment by glours** [comment]
@thaJeztah I opened this PR as I was able to update the one opened by the bot.
You could see [the error](https://github.com/docker/docs/actions/runs/4136332532/jobs/7150038450#step:4:3582) I mentioned to you.
Feel free to update directly this PR
@ndeloof if you can follow that one during my time off please 🙏

**Comment by thaJeztah** [comment]
I'm looking at the failure;

```
#15 40.02   Liquid Exception: undefined method `gsub' for nil:NilClass in engine/reference/commandline/compose_config.md
#15 40.02                     ------------------------------------------------
#15 40.02       Jekyll 4.2.2   Please append `--trace` to the `build` command 
#15 40.02                      for any additional information or backtrace. 
#15 40.02                     ------------------------------------------------
#15 40.02 /src/_plugins/relative_links_filter.rb:47:in `replace_relative_links': undefined method `gsub' for nil:NilClass (NoMethodError)
```

File: _redirects.yml [hunk 1]
```diff
@@ -128,7 +128,8 @@
 "/desktop/windows/wsl/":
   # Link used by Docker Desktop to refer users on how to activate WSL 2
   - /go/wsl2/
-
+"/engine/reference/commandline/compose_config/":
+  - /engine/reference/commandline/compose_convert/
```
Comment (thaJeztah) [inline _redirects.yml:132]: Not the cause of the error, but for these, we usually use the `redirects_from` metadata in the target markdown file. That way the redirects get preserved when the file moves around, which can make it easier to maintain.

**Comment by thaJeztah** [comment]
Ah! I think I see what's happening; I'll push a commit.

**Comment by thaJeztah** [comment]
@glours this was the issue; the stub has a reference to the name of the datafile (yaml) to include, so that had to be updated. I also moved the redirect to that file; https://github.com/docker/docs/compare/9d77a4495080202daeb66b365ec12e6687a9f7a0..76a295426016557f883f9775c61c71ab16f8e9db
Screenshot 2023-02-10 at 09 31 24

**Review by ndeloof — APPROVED** [review]

**Review by aevesdocker — APPROVED** [review]
Thanks @glours and thank you @thaJeztah for your help with this!

**Comment by glours** [comment]
Thanks @thaJeztah for the fix and explanation, I'll know for the next time 😉

**Pull request merged**